sim/src/sim_element.rs

```diff
@@ -1,11 +1,20 @@
+//! A sim_element is something that takes in RPCs and give them to other sim_elements.
+//! Right now the only sim_elements are nodes, edges, and plugin_wrappers.
+
 use rpc_lib::rpc::Rpc;
 
 pub trait SimElement {
-    fn add_connection(&mut self, neighbor: u32);
+    fn add_connection(&mut self, neighbor: String);
 
-    fn tick(&mut self, tick: u64) -> Vec<(Rpc, Option<u32>)>;
+    fn tick(&mut self, tick: u64) -> Vec<(Rpc, Option<String>)>;
 
-    fn recv(&mut self, rpc: Rpc, tick: u64, sender: u32);
+    fn recv(&mut self, rpc: Rpc, tick: u64, sender: &str);
```

Comment:
Is there any reasoning behind these two functions in particular taking &str arguments instead of Strings?  Or is it just the configuration that allows the least amount of copying?

Reply:
Mostly convenience. Instead of having to call to_string() every time you pass a string into this call, you directly convert at the source. Makes the code a bit more readable. Ultimately, the copying should only be done when the object itself is initialized and owns the string.

Reply:
Okay, that makes sense.